src/Pages/SignupPage/Components/SignupForm.jsx

```diff
@@ -0,0 +1,236 @@
+  };
+
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  const { authState, authDispatch } = useAuth();
+  const [signupDetails, setSignupDetails] = useState(intialState);
+  const { showAlert, setShowAlert } = useAlert();
+  const [showPassword, setShowPassword] = useState({
+    password: false,
+    confirmPassword: false,
+  });
+  const [error, setError] = useState(initialErrorState);
+  const { setShowModal } = useModal();
+  const signupInputChangeHandler = (event) => {
+    setSignupDetails({
+      ...signupDetails,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const showPasswordHandler = (key) => {
+    setShowPassword({
+      ...showPassword,
+      [key]: !showPassword[key],
+    });
+  };
+
+  const submitHandler = async (event) => {
+    event.preventDefault();
+    const { firstName, lastName, email, password, confirmPassword } =
+      signupDetails;
+    setError(initialErrorState);
+
+    if (password.length < 6) {
+      setError({
+        error: true,
+        message: "Password must be at least 6 characters long",
+        passwordError: true,
+      });
+      setSignupDetails({ ...signupDetails, password: "", confirmPassword: "" });
+      return;
+    }
+
+    if (password.search(/\d/) === -1) {
+      setError({
+        error: true,
+        message: "Password must contain at least one number",
+        passwordError: true,
+      });
+      setSignupDetails({ ...signupDetails, password: "", confirmPassword: "" });
+      return;
+    }
+
+    if (password.search(/[a-z]/) === -1) {
+      setError({
+        error: true,
+        message: "Password must contain at least one lowercase letter",
+        passwordError: true,
+      });
+      setSignupDetails({ ...signupDetails, password: "", confirmPassword: "" });
+      return;
+    }
+
+    if (password.search(/[A-Z]/) === -1) {
+      setError({
+        error: true,
+        passwordError: true,
+        message: "Password must contain at least one Uppercase letter",
+      });
+      setSignupDetails({ ...signupDetails, password: "", confirmPassword: "" });
+      return;
+    }
+
+    if (password !== confirmPassword) {
+      setError({
+        error: true,
+        message: "Passwords do not match",
+        confirmPasswordError: true,
+        passwordError: true,
+      });
+      setSignupDetails({ ...signupDetails, password: "", confirmPassword: "" });
+      return;
+    }
+
+    try {
+      const response = await axios.post("/api/auth/signup", {
+        firstName,
+        lastName,
+        email,
+        password,
+      });
+      if (response.status === 201) {
+        setSignupDetails(intialState);
+        authDispatch({ type: "SIGNUP", payload: response.data });
+        setShowAlert({
+          showAlert: true,
+          alertMessage: "Account created successfully!",
+          type: "success",
+        });
+        if (pathname === "/signup") {
+          navigate(-2);
+        }
```

Comment:
What is the use navigate(-2)

Reply:
It is helping me go back two routes back in history. Actually I wanted to check path in history before navigating but I still do not know how. So user is landing on signup from login page so this is just a quick hack to land him back on the page he was on prior to login.